### PR TITLE
OCPBUGSM-32117: Adding ipv6 support to dns busy address workaround

### PR DESCRIPTION
**Pull request by tsorya**
We have a workaround that deletes service that took address of dns
service. Till now it supported only ipv4. This change adds ipv6 support

### CODE REVIEW

File: src/assisted_installer_controller/assisted_installer_controller.go [hunk 1]
```diff
@@ -254,13 +254,14 @@ func (c *controller) HackDNSAddressConflict(wg *sync.WaitGroup) {
 		return
 	}
 
-	ip, _, _ := net.ParseCIDR(networks[0])
-	ip4 := ip.To4()
-	if ip4 == nil {
-		c.log.Infof("Service network is IPv6: %s, skipping the .10 address hack", ip)
+	netIp, _, _ := net.ParseCIDR(networks[0])
+	ip := netIp.To16()
```
Comment (eranco74) [inline src/assisted_installer_controller/assisted_installer_controller.go:258]: :candy:

**Review by eranco74 — COMMENTED** [review]
Nice!
/lgtm

File: src/assisted_installer_controller/assisted_installer_controller.go [hunk 2]
```diff
@@ -254,13 +254,14 @@ func (c *controller) HackDNSAddressConflict(wg *sync.WaitGroup) {
...
+	if ip == nil {
+		c.log.Infof("Failed to parse service network cidr %s, skipping", networks[0])
 		return
 	}
-	ip4[3] = 10 // .10 is the conflicting address
+
+	ip[len(ip)-1] = 10 // .10 is the conflicting address
```
Comment (empovit) [inline src/assisted_installer_controller/assisted_installer_controller.go:264]: fix the comment, it's `.10` or `:a`, or just 10 at the last position

File: src/assisted_installer_controller/assisted_installer_controller_test.go [hunk 3]
```diff
@@ -1417,6 +1421,10 @@ var _ = Describe("installer HostRoleMaster role", func() {
 		})
 		It("Exit if service network is IPv6", func() {
```
Comment (empovit) [inline src/assisted_installer_controller/assisted_installer_controller_test.go:1422]: Rename the test to reflect the new functionality

**Pull request merged**